src/utils/ImmutableUtil.js

```diff
@@ -3,7 +3,7 @@ import Immutable from 'immutable'
 function immutable(StoreModel) {
   StoreModel.config = {
     setState(currentState, nextState) {
-      this.state = nextState
+      this.state = currentState.merge(nextState)
```

Comment:
this works well but then limits us to only being able to use Map

cc @jareware